File: 프로그래머스/lv2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    let check = Array.from({length:numbers.length},()=>0)
    let set = new Set();
    
    const DFS = (str)=>{
        if(+str > 1) set.add(+str);
        
        for(let i=0; i<numbers.length; i++){
            if(check[i]===0){
                check[i] = 1;
                DFS(str+numbers[i]);
                check[i] = 0;
            }
        }
    }
    
    const 소수판별 = (num)=>{
        let result = true;
        for(let i=2; i<=Math.sqrt(num); i++){
            if(num%i===0) result = false;
        }
        return result;
    }
    
    DFS('')
    
    return [...set].filter(number=> 소수판별(number)).length;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    
    HashSet<Integer> set = new HashSet<>();
    
    public int solution(String numbers) {
        recursive("",numbers);
    
        return (int) set.stream().filter(i-> isPrime(i)).count();
    }
    
    void recursive(String comb, String others){
        if(!comb.equals("")){
            set.add(Integer.parseInt(comb));
        }
        for(int i=0; i<others.length(); i++){
            recursive(comb+others.charAt(i),others.substring(0,i)+others.substring(i+1));
        }
    }
    boolean isPrime(int num){
        if(num == 1 || num == 0) return false;
        
        for(int i=2; i <= (int) Math.sqrt(num); i++){
            if(num % i == 0){
                return false;
            }
        }
        return true;
    }
}